### PR TITLE
[TIR] Bugfix for zero number arguments tir functions.

### PR DESCRIPTION
**Pull request by ZihengJiang**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

File: tests/python/unittest/test_tir_base.py [hunk 1]
```diff
@@ -36,11 +36,22 @@ def test_scalar_add():
     c = tir.ret(c)
     c = tir.Evaluate(c)
     func = tir.PrimFunc([a, b], c)
+    print(func)
```
Comment (tqchen) [inline tests/python/unittest/test_tir_base.py:39]: remove prints

File: src/tir/transforms/make_packed_api.cc [hunk 2]
```diff
@@ -120,6 +120,11 @@ PrimFunc MakePackedAPI(PrimFunc&& func, int num_unpacked_args) {
   int num_args = static_cast<int>(func_ptr->params.size());
   ICHECK_LE(num_unpacked_args, num_args);
 
+  bool pack_args = true;
+  if (num_unpacked_args == -1) {
```
Comment (tqchen) [inline src/tir/transforms/make_packed_api.cc:124]: Can we also update the python side and update the documents about this behavior>

**Review by tqchen — CHANGES_REQUESTED** [review]

File: src/tir/transforms/make_packed_api.cc [hunk 3]
```diff
@@ -120,6 +120,11 @@ PrimFunc MakePackedAPI(PrimFunc&& func, int num_unpacked_args) {
...
+    pack_args = false;
+    num_unpacked_args = 0;
+  }
   int num_packed_args = num_args - num_unpacked_args;
```
Comment (areusch) [inline src/tir/transforms/make_packed_api.cc:128]: seems like we could `ICHECK_GE(num_unpacked_args, 0)` here now

**Review by areusch — COMMENTED** [review]
@ZihengJiang could you say more about the issue that necessitates this bugfix? perhaps a traceback or something?

File: src/tir/transforms/make_packed_api.cc (hunk 3 above)
Comment (ZihengJiang) [reply src/tir/transforms/make_packed_api.cc:128]: Updated

**Comment by ZihengJiang** [comment]
Hi @areusch ! The motivation of the PR is here: https://discuss.tvm.apache.org/t/tvm-primfunc-returning-a-constant-cannot-compile/10577/2
It is mainly because we cannot distinguish to pack/unpack in the case of `num_args == num_unpacked_args == 0` with previous logic.

File: python/tvm/tir/transform/transform.py [hunk 4]
```diff
@@ -331,14 +331,15 @@ def LowerCustomDatatypes():
...
+def MakePackedAPI(num_unpacked_params: int = -1):
     """Transform the PrimFuncs in the module to a packed func API.
 
     Parameters
     ----------
     num_unpacked_params : int
         Number of parameters that we hope to directly pass via normal arguments
-        following the PackedFunc input signature.
+        following the PackedFunc input signature. If it is specified as -1 or it
+        is less than the number of arguments, the passwill packed arguments still.
```
Comment (junrushao1994) [inline python/tvm/tir/transform/transform.py:342]: nitpick

```suggestion
        is less than the number of arguments, the pass will packed arguments still.
```

**Review by tqchen — APPROVED** [review]

**Pull request merged**